demo/site/next.config.js

```diff
@@ -19,6 +19,20 @@ const nextConfig = {
             },
         ];
     },
+    async redirects() {
+        const adminUrl = process.env.ADMIN_URL;
+
+        if (!adminUrl) {
+            throw Error("ADMIN_URL is not defined");
+        }
+        return [
+            {
+                source: "/admin",
+                destination: adminUrl,
+                permanent: false,
+            },
+        ];
+    },
```

Comment:
I moved the /admin redirect from middleware into config as that was the only remaining redirect and is easier implemented here